### ISSUE TITLE
Should toggle-modules run before config-import in sync:refresh?

### ISSUE THREAD

**Issue by rjgwiz** [acquia/blt#2671]
`toggle-modules` was added to `sync:refresh` in #2342. It runs after `setup:config-import`. Is there any reason `toggle-modules` cannot or should not run first? 

The benefit is increased efficiency by not running update hooks for modules that will be uninstalled soon after. And, never having to worry about any update hooks erroring for a module that shouldn't run in that environment anyway (could error due to config split, third party integrations not available for that environment, etc.).

Are there any drawbacks or blockers to running `toggle-modules` first?

**Comment by grasmash** [comment]
Yeah that'd be problematic. Toggle modules is intended to be able to override exported config. That wouldn't be possible if you ran it before importing config.

**Comment by rjgwiz** [comment]
@malikkotob I don't have extensive experience with `config-import`, would that scenario occur if `dblog` was blacklisted and no non-prod environment config split was put in place for `dblog`?

**Comment by rjgwiz** [comment]
I think I glossed over the core config management import (overlooked since I never use it). This definitely wouldn't play nice with that.

**Comment by wu-edward** [comment]
@rjgwiz Typically if you're using config-split, you use the blacklists to "toggle" modules you want enabled/disabled on each environment. `toggle-modules` is more for use with a Features-based workflow.

**Comment by mikemadison13** [comment]
if you absolutely had to, you could always turn a module on via an update hook so it happened well ahead of any configuration imports, but agreed with @wu-edward that a config split method would likely be a little "better"

**Comment by rjgwiz** [comment]
We are using a primarily features-based workflow. I can replace `sync:refresh` [with a custom command](https://blt.readthedocs.io/en/stable/readme/extending-blt/) to swap the order of `toggle-modules` and `setup:config-import` though maintaining an override isn't ideal. 

I created this issue to do a pulse check and so far it seems to be a unique issue. The only idea that comes to mind of how to accommodate this on a BLT level for all config workflows is to read or allow an override of the list of commands invoked in `sync:refresh` in `project.yml`. That doesn't sound desirable in itself, for many reasons, and doesn't seem worth pursuing to address an uncommon issue.

**Comment by malikkotob** [comment]
@rjgwiz:
> would that scenario occur if dblog was blacklisted and no non-prod environment config split was put in place for dblog?

It would not, but as @wu-edward mentioned, `toggle-modules` is more so for use with a features-based workflow, as the blacklists in the`config-split` flow circumvent the need for it.

The `toggle-modules` feature, as it stands, is in place to _override_ exported config, so executing it before-hand negates the feature. Swapping the order would result in something like the following happening for a local environment where the `shield` module, for example, should be disabled:
- `toggleModules` runs and `shield` is disabled
- `config-import` runs and `shield` is re-enabled

If there are specific modules that are causing issues that need disabling prior to the update hooks running, you can write a custom hook that executes prior to the config import.
